### PR TITLE
Backend time availability

### PR DESCRIPTION
**Pull request by ZhenyuYu1**
New features:
1. Added column to attendees table for time segment availability
2. Time segments is a JSON in the form of: {
  "Aug 10": [],
  "Aug 11": [
    {
      "beginning": "09:00",
      "end": "10:00",
      "type": "Preferred"
    },
    {
      "beginning": "11:00",
      "end": "12:00",
      "type": "Regular"
    }
  ],
  "Aug 12": [],
  "Aug 15": []
}
3. Can store by either days of the week or specific days
4. Can retrieve data to get a user and their associated time segments for a given event

### CODE REVIEW

**Comment by pselvana** [comment]
Please run prettier :)

Also, for API's to work off of CloudFlare, we need to have them as edge functions.  You can't see the logs in cloudflare unfortunately but here's the relevant lines:

10:32:20.932 | ⚡️ ERROR: Failed to produce a Cloudflare Pages build from the project.
-- | --
10:32:20.932 | ⚡️
10:32:20.933 | ⚡️ 	The following routes were not configured to run with the Edge Runtime:
10:32:20.933 | ⚡️ 	  - /api/attendees/create
10:32:20.933 | ⚡️ 	  - /api/attendees
10:32:20.934 | ⚡️
10:32:20.934 | ⚡️ 	Please make sure that all your non-static routes export the following edge runtime route segment config:
10:32:20.934 | ⚡️ 	  export const runtime = 'edge';
10:32:20.934 | ⚡️
10:32:20.934 | ⚡️ 	You can read more about the Edge Runtime on the Next.js documentation:
10:32:20.935 | ⚡️ 	  https://nextjs.org/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes


The fix is as simple as adding that export in the two files.  You can see examples in /api/user/create
@AlanaDubie @Jean-LucMartel - Mentioning you here because I forgot to mention this requirement for deploying.

File: src/app/api/attendees/create/route.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+export async function POST(request: Request) {
+  const cookieStore = cookies()
+  const supabase = createServerClient(cookieStore)
+  const body = await request.json()
+
+  const { data, error } = await supabase
+    .from('attendees')
+    .insert([body])
+    .select()
```
Comment (pselvana) [inline src/app/api/attendees/create/route.ts]: I'm not sure why the select is here. Is it necessary?

File: src/app/test/page.tsx [hunk 2]
```diff
@@ -1,43 +1,55 @@
 'use client'
 
-import { UUID } from 'crypto'
 import { useEffect, useState } from 'react'
-import { createUser, getUser } from '@/utils/userUtils'
+import { getAttendees, addAttendee } from '@/utils/attendeesUtils'
+import { UUID } from 'crypto'
+
+interface Attendee {
```
Comment (pselvana) [inline src/app/test/page.tsx]: can this be in the atteneesUtils for re-use?

File: src/app/test/page.tsx [hunk 3]
```diff
@@ -1,43 +1,55 @@
 'use client'
 
-import { UUID } from 'crypto'
 import { useEffect, useState } from 'react'
-import { createUser, getUser } from '@/utils/userUtils'
+import { getAttendees, addAttendee } from '@/utils/attendeesUtils'
```
Comment (pselvana) [inline src/app/test/page.tsx]: Is this file needed? "test"

**Review by pselvana — CHANGES_REQUESTED** [review]
Fix up the build please. Some light changes to consider.

File: src/app/test/page.tsx (hunk 2 above)
Comment (ZhenyuYu1) [reply src/app/test/page.tsx]: Moved Attendee interface to attendeeUtils.

File: src/app/api/attendees/create/route.ts (hunk 1 above)
Comment (ZhenyuYu1) [reply src/app/api/attendees/create/route.ts]: Removed select.

File: src/app/test/page.tsx (hunk 3 above)
Comment (ZhenyuYu1) [reply src/app/test/page.tsx]: Deleted test file.

**Review by pselvana — APPROVED** [review]
LGTM!
